src/coreclr/tools/aot/ObjWriter/objwriter.cpp

```diff
@@ -495,15 +491,23 @@ void ObjectWriter::EmitWinFrameInfo(const char *FunctionName, int StartOffset,
   const MCExpr *BaseRefRel =
       GetSymbolRefExpr(FunctionName, MCSymbolRefExpr::VK_COFF_IMGREL32);
 
+  Triple::ArchType Arch = TMachine->getTargetTriple().getArch();
+
+  if (Arch == Triple::thumb || Arch == Triple::thumbeb) {
+    StartOffset |= 1;
+  }
+
   // start Offset
   const MCExpr *StartOfs = MCConstantExpr::create(StartOffset, *OutContext);
   Streamer->EmitValue(
       MCBinaryExpr::createAdd(BaseRefRel, StartOfs, *OutContext), 4);
 
-  // end Offset
-  const MCExpr *EndOfs = MCConstantExpr::create(EndOffset, *OutContext);
-  Streamer->EmitValue(MCBinaryExpr::createAdd(BaseRefRel, EndOfs, *OutContext),
-                      4);
+  if (Arch == Triple::x86 || Arch == Triple::x86_64) {
+    // end Offset
+    const MCExpr *EndOfs = MCConstantExpr::create(EndOffset, *OutContext);
+    Streamer->EmitValue(
+        MCBinaryExpr::createAdd(BaseRefRel, EndOfs, *OutContext), 4);
```

Comment:
What is this space for on x86/x64?

Reply:
Here we emit an `IMAGE_RUNTIME_FUNCTION_ENTRY` record. For x86/x64 it contains `BeginAddress`, `EndAddress`, and `union { UnwindInfoAddress; UnwindData; }`. For ARM/ARM64 it contains just `BeginAddress` and `UnwindData`.